tests/transforms/test_batch_params.py

```diff
@@ -27,20 +27,22 @@ def test_simple_circuit(mocker):
 
     @qml.batch_params
     @qml.qnode(dev)
-    def circuit(data, x, weights):
+    def circuit(data, data2, x, weights):
         qml.templates.AmplitudeEmbedding(data, wires=[0, 1, 2], normalize=True)
+        qml.templates.AngleEmbedding(data2, wires=[0, 1, 2])

```

Comment:
Two thoughts here:

1. This is a great way of testing that the functionality works :+1: 
2. Likely, it will be good to have a general way of writing test cases to verify the batching support of new templates. With the current approach, if every template is added to this circuit, then it will grow into a complicated circuit (rather than a simple one).

For 2. there could be two potential ways to solve the issue:
1. Creating a separate test case just for `qml.templates.AngleEmbedding`
2. [Parametrizing the test case](https://docs.pytest.org/en/6.2.x/parametrize.html) using `pytest.mark.parametrize`: the parametrized input parameters could for example be the embeddings that are tested (for now). Likely we'd need to create a parameter for:
* The type of embedding being tested (`qml.templates.AmplitudeEmbedding`, `qml.templates.AngleEmbedding`);
* The input shape required by the template (e.g., `2 ** num_wires` for amplitude, `num_wires` for angle embedding);
* Potentially the wires (though could be added later on too, as for now both existing embeddings use the same wires).